### PR TITLE
docs: cruise wins over yolo — correct four sources that said otherwise

### PR DESCRIPTION
**Pull request by arbeithand**
Maintainer decision: when both `fabrik:cruise` and `fabrik:yolo` are present, **cruise takes precedence**. Cruise is the conservative label — adding yolo to a cruise issue should never widen automation.

That is also what the code has always done, and what two tests have been asserting all along:

```
TestHandleStageComplete_BothCruiseAndYolo_CruiseWins  handle_stage_complete_test.go:499
TestCruiseCatchUp_BothCruiseAndYolo_CruiseWins        poll_test.go:1192
```

Both pass on `main` today. The docs have been contradicting green tests.

## What was wrong

**Wrong outright** — claimed "yolo takes precedence":
- `CLAUDE.md`
- `specs/898-cruise-full-pipeline-auto/spec.md` (edge cases + out of scope)

**Half right** — conceded the merge decision but claimed yolo overrides stop-at-Validate, so the issue *"still advances to Done"*:
- `docs/USER_GUIDE.md` (3 places)
- `docs/state-machine.md` (label table, label semantics reference)

Neither holds. Both end-of-Validate guards test the **raw** cruise label, not `cruiseActive`:

```go
// engine/stages.go
if hasCruiseLabel(item) && stage.Name == "Validate" { shouldAdvance = false }
if hasCruiseLabel(item) { return ... }   // attemptMergeOnValidate
```

So cruise wins the merge decision **and** the advance-to-Done decision. The issue stops at Validate exactly as it would with cruise alone.

## Also fixed

The comment above `cruiseActive` asserted *"cruise is suppressed so yolo always takes precedence"* — eight lines above a guard that does the opposite. `cruiseActive` is false under yolo only to avoid double-counting in the `shouldAdvance` expression, where `yoloActive` already contributes the same value at every non-Validate stage. It does not express precedence, and reading it that way is what made the surrounding docs plausible.

## Provenance

Found by the canonical-docs audit as finding **L604**, which correctly classified it as an *intent conflict requiring a maintainer decision* rather than drift — the code and the spec genuinely disagreed, so there was no way to tell which was authoritative without asking.

No behaviour change; `docs/llms-full.txt` regenerated in the same commit.

### CODE REVIEW

**Comment by arbeithand** [comment]
Addressed in 496dc358 — a correct catch, and the diagnosis was exact.

`docs/state-machine.md` §"shouldAdvance resolution (Path 1)" step 5 said `cruiseActive && stage.Name == "Validate"`; the code uses the raw label. Since step 2 defines `cruiseActive = !yoloActive && hasCruiseLabel(item)`, the derivation implied the guard does not fire under yolo — reintroducing, in step-by-step form, the exact claim the rest of the PR removes, and contradicting line 299 of the same file.

Both steps now fixed:

- **Step 5** → `hasCruiseLabel(item) && stage.Name == "Validate"`, noting explicitly that it is the raw label and therefore fires even when yolo is present.
- **Step 2** → now states that suppressing `cruiseActive` under yolo is *not* a precedence rule; it only avoids double-counting in step 3, where `yoloActive` already contributes the same `true`.

`docs/llms-full.txt` regenerated — it had carried the stale line verbatim, as you noted.

**On the root cause**, since it is worth recording: the first pass grepped for the *prose* forms of the claim (`yolo takes precedence`, `still advances to Done`, `overridden by yolo`) and never searched for the claim expressed as a *code expression*. A derivation section restates the same rule in a different notation, so a prose-only sweep structurally cannot find it. `grep -rn "cruiseActive" docs/` would have — that is the check I should have run, and it is now the one that verifies the fix.

That failure mode generalises beyond this PR: the canonical-docs audit found ~20 places where the document contradicts itself, and the wrong half is nearly always a *summary or derivation* restating a rule the normative section gets right. Prose-keyed sweeps miss precisely those.

File: specs/898-cruise-full-pipeline-auto/spec.md [hunk 1]
```diff
@@ -92,7 +92,7 @@ After cruise stops at Validate-complete with an open PR, a human (or the test ha
 
 ## Out of Scope
 
-- Verifying the yolo+cruise coexistence edge case end-to-end (yolo takes precedence; covered by unit tests in `engine/stages_test.go`).
+- Verifying the yolo+cruise coexistence edge case end-to-end (cruise takes precedence; covered by unit tests in `engine/stages_test.go`).
```
[the remark this answers is not in the document]
Comment (arbeithand) [reply specs/898-cruise-full-pipeline-auto/spec.md]: Correct — fixed in 1f66044c.

`engine/stages_test.go` has only `TestAttemptMergeOnValidate_CruiseSkipsAutoMerge`, which exercises cruise **alone** (no yolo label), so it does not cover the coexistence case at all. The line now cites the two tests that do pin it, one per advancement path:

- `TestHandleStageComplete_BothCruiseAndYolo_CruiseWins` (`engine/handle_stage_complete_test.go:499`) — Path 1, `handleStageComplete`
- `TestCruiseCatchUp_BothCruiseAndYolo_CruiseWins` (`engine/poll_test.go:1192`) — Path 2, catch-up loop

and notes explicitly what `stages_test.go` does cover, so the next reader is not sent there again. All three citations verified present by name before pushing.

Cause: the first pass rewrote the *claim* on that line ("yolo takes precedence" → "cruise takes precedence") and left the stale citation sitting beside it. Same shape as the `cruiseActive` miss you caught in the previous round — I corrected what my search pattern matched and did not re-read the rest of the sentence.

**Comment by arbeithand** [comment]
Both addressed in 284cafe3. (Responding here rather than in-thread: these were "could not anchor to diff" findings, so there is no review thread to reply to or resolve — the one inline thread on this PR was resolved in the previous round.)

Verified both, and they were worse than off-by-a-few-lines:

| citation | actually points at |
|---|---|
| `engine/poll.go:1410–1444` | `selectDeepFetchCandidates` internals — the cruise gate is not in `poll.go` at all |
| `engine/stages_test.go:60–75` | `TestAttemptMergeOnValidate_UnresolvedCurrentHeadThread_Defers` (:68), unrelated; the cruise test is at :184 |

**I did not re-pin them.** This section had rotted twice, so re-pinning just resets the clock. Citations now name symbols. A line number is invalidated by any insertion above it and keeps pointing confidently at whatever now occupies that offset — it fails *silently*, which is precisely why both of these survived until you read them. A symbol name survives edits above it and fails loudly: `grep` returns nothing. This follows recommendation (a) of the canonical-docs audit ("stop pinning line numbers in as-built docs; cite symbol names only"), and I verified every symbol now cited in the spec resolves before pushing.

**One substantive correction beyond the citations.** The old line described *"the code gate"* — singular. There are two, and both key on the raw `hasCruiseLabel`:

- `handleStageComplete` — `hasCruiseLabel(item) && stage.Name == "Validate"` suppresses the **Done advance**
- `attemptMergeOnValidate` — early return suppresses **auto-merge**

Citing only one of them is the exact reading error that produced the "yolo overrides cruise's stop-at-Validate" claim this PR exists to remove. So the stale citation was not merely cosmetic here — it was load-bearing for the misconception.

Three rounds, three real findings, all mine. Worth noting the pattern for the record: each time I corrected what my search pattern matched and stopped there — prose claims but not the code-expression restatement, the claim on a line but not the citation beside it, and now the citation but not the section around it.
